common/flatpak-installation.c

```diff
@@ -3074,6 +3074,56 @@ flatpak_installation_list_unused_refs (FlatpakInstallation *self,
                                        const char          *arch,
                                        GCancellable        *cancellable,
                                        GError             **error)
+{
+  return flatpak_installation_list_unused_refs_with_options (self, arch, NULL, cancellable, error);
+}
+
+static void
+prune_excluded_refs (char **full_refs,
+                     char **refs_to_exclude)
+{
+  guint len = g_strv_length (full_refs);
+  for (guint i = 0; i < len; i++)
+    {
+      if (g_strv_contains ((const gchar * const*)refs_to_exclude, full_refs[i]))
+        {
+          g_free (full_refs[i]);
+          full_refs[i] = full_refs[len - 1];
+          full_refs[len - 1] = NULL;
+          len--;
```

Comment:
I think you also need to i-- here, because we need to re-check the full_refs[i] which you just switched in.